datafusion/src/physical_plan/expressions/array_agg.rs

```diff
@@ -120,7 +120,17 @@ impl Accumulator for ArrayAggAccumulator {
     }
 
     fn merge(&mut self, states: &[ScalarValue]) -> Result<()> {
-        self.update(states)
+        if states.is_empty() {
+            return Ok(());
+        };
+
+        match &states[0] {
```

Comment:
I think it would makes sense to assert here that `states.len() == 1`? so we don't (silently) end up ignoring any other items that might be added (accidentally)

Reply:
yea, let me add it.